### DISCUSSION TITLE
Allow users to invite other users

### DISCUSSION THREAD

**Discussion by silkgh** [supabase/supabase#30569] [Questions] [answered]
I'm scoping out the use cases I need to prove in my POC (for supabase as our choice of backend)

Basically, I need to allow a trainer to invite their own clients via the app ; all they'd do is fill in an email address field and hit "send".

The email that's sent out has to 1, be branded to my company and 2, provide a link to download the app and within that link pass a trainer_id param to link to the new user once they download then sign up. All I can find is this: https://supabase.com/docs/reference/javascript/auth-admin-inviteuserbyemail - which doesn't provide much detailed info.

I also want to run a referrals programme where a trainer can invite other trainers - would work in a similar way to the above.

I know this is a fairly common use case so I'm just looking for an answer on "is this possible with supabase auth" and any  pointers if so. Likewise if it isn't possible, any advice on other auth SDKs that link up well with supabase & powersync would be welcome.

**Comment by GaryAustin1** [comment, marked as answer]
You probably would need to look at that admin auth call or the other one below,  server side or with edge functions.  There is no built in tool to do this.

If you customize your Supabase email templates you could use the call you show and add links to possibly do what you want in those templates.  

If you want to generate your own emails with more complexity or variety for different cases then you could use https://supabase.com/docs/reference/javascript/auth-admin-generatelink to get the needed info for those emails.  You would send them on your own to a SMTP provider.
